src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1597,8 +1599,28 @@ private static bool IsAncestor(DependencyObject? instance, AncestorsDictionary m
 					if (instance == ancestor)
 					{
 						isAncestor = true;
+						break;
+					}
+				}
+
+				if (iterations < 1)
+				{
+					var level =
+#if DEBUG
+						// Make layout cycles visible in debug builds, until Children.Add
+						// validates for cycles.
+						LogLevel.Error;
```

Comment:
Should we even throw?

Reply:
It's not something that we can troubleshoot easily at this point, maybe with #1621.

Reply:
Nope... that's an error, but here it should just return false.